src/Orchard/Security/Providers/FormsAuthenticationService.cs

```diff
@@ -35,62 +39,32 @@ public class FormsAuthenticationService : IAuthenticationService {
             IMembershipService membershipService,
             IHttpContextAccessor httpContextAccessor,
             ISslSettingsProvider sslSettingsProvider,
-            IMembershipValidationService membershipValidationService) {
+            IMembershipValidationService membershipValidationService,
+            IEnumerable<IUserDataProvider> userDataProviders,
+            ISecurityService securityService) {
+
             _settings = settings;
             _clock = clock;
             _membershipService = membershipService;
             _httpContextAccessor = httpContextAccessor;
             _sslSettingsProvider = sslSettingsProvider;
             _membershipValidationService = membershipValidationService;
+            _userDataProviders = userDataProviders;
+            _securityService = securityService;
 
             Logger = NullLogger.Instance;
 
-            ExpirationTimeSpan = TimeSpan.FromDays(30);
         }
 
         public ILogger Logger { get; set; }
 
-        public TimeSpan ExpirationTimeSpan { get; set; }
+        public TimeSpan ExpirationTimeSpan {
```

Comment:
This is breaking what users currently have to do to change this value (host.config).
Not sure it's wise to make it editable in the UI. More code, not usually used, accessible to more users, ... lots of reasons to keep it the way it is today.

Reply:
The value for this previously was contant and 30 days. I don't see where it was being read from host.config. Unless writing the form authentication settings there directly overrides anything that is done in code here for the auth cookie.

In this PR, the value is in a setting that is editable from the UI (that is the implementation I put in Orchard.Users, not the default one from Orchard.Framework, even though I realize that close to 100% tenants have Orchard.Users enabled), but here in the service we are getting it from a service, rather than from the setting. It's straightforward to write a service that reads from a config file, or that does something else entirely. DIfferent calls could lead to different values for this.

Reply:
The value is assigned automatically by autofac when the service is resolve. It's called "Property Injection". I don't think we need a service for this.

Reply:
I did not know that.
I am trying now to see how to set this up in a per-tenant way using the .config files ("Sites.MyTenant.config", right?)

Reply:
We could have both the services and property injection:

If I moved that `ExpirationTimeSpan` to DefaultSecurityService I could still do Property Injection into there.

Alternatively, in the constructor of `FormsAuthenticationService` I could initialize the value of `ExpirationTimeSpan` to a negative value. Then rather than using the property in the rest of the class, I could have a method like:
```
public TimeSpan GetExpirationTime() {
            if (ExpirationTimeSpan < TimeSpan.Zero) {
                // We have not set expiration from Sites.Tenant.config
                return  _securityService.GetAuthenticationCookieLifeSpan();
            }
            // we have set the expiration from the config file
            return ExpirationTimeSpan;
        }
```
As long as the UI for each of those services clearly stated that having a value in the config file would take precedence, that behaviour would not be obscure.

Reply:
The question, I guess, is more of a "should we" than a "could we" nature, at this point, right?

Reply:
Personally I think there may be value in having the option, for example, to set different `ExpirationTimeSpan` depending on the "source" of the SignIn. For example, if you SignIn from a browser I may set a short validity, but if you SignIn from my app on your phone I may decide that that may be ok with a longer-lived cookie (of course that requires me to identify the source in my service, but that is a separate issue).

Reply:
Another scenario may be:
 - Cookie expires in a bunch of days for admin
 - Cookie expires in a shorter timeframe for "normal" users